src/actions.sol

```diff
@@ -92,6 +92,9 @@ contract Actions is DSNote {
     }
 
     function repay(ShelfLike shelf, ERC20Like erc20, uint loan, uint amount) public {
+        uint debt = PileLike(shelf.pile()).debt(loan);
+        require(amount <= debt, "amount-larger-than-debt");
```

Comment:
Instead of reverting we could also say:
```javascript
if(amount > debt) {
   amount = debt;
}
```
I think in most of the cases the user actually wants to repay, so we can try to make it possible.
wdyt?

Reply:
That's already happening in the shelf, afaik the idea was to actually revert here such that it's not even possible, to prevent currency getting stuck in the proxy.